### ISSUE TITLE
Status of entity having different values on multiple devices

### ISSUE THREAD

**Issue by fraankkr** [Mat931/digitalstrom-homeassistant#14]
Hi Mat

I found out while using your digitalstrom-homeassistent "binding" that I have different values of an entity on diffrent devices. For example when switching on a lamp in the living area it is changing the status from off to on on the laptop but still has the status "off" on my mobile phone. 
The other way around I have found the same issue. Switching on or off other lamps or devices does not sync to other mobile phones as well ? Any idea what the reason could be ?

I also found out, that sometimes different devices are not reachable due connection errors. could it be the reason that my digitalstrom -server is getting old ? Using a first genereation dss11-1GB.

Thanks for your reply and awesome work btw.

F

**Comment by Mat931** [comment]
Hi @fraankkr !

Are you using Home Assistant or digitalSTROM on your different devices? When switching lights with digitalSTROM it takes a while until Home Assistant sees the new state. This problem and a potential workaround I can try to fix it was discussed here: https://github.com/Mat931/digitalstrom-homeassistant/issues/10

How do you see that devices are not reachable? Are they just unavailable in Home Assistant or are there any errors in the log? My integration currently only checks the availability of a device once when it is loaded, that is maybe something I should fix. Does a reload of the integration make the device available again?

If you want to reload the integration or generate a debug log you can do that [here](https://my.home-assistant.io/redirect/integration/?domain=digitalstrom).

Thanks!
Mat

**Comment by fraankkr** [comment]
Hi Mat

Thanks for your great work again and your fast response. I will attach a short video file for your futher investigation. To make it short you will find three things there:

1. Switching on Wohnzimmerlampe will only change the status in one section, but not in the other (right in the beginning of the video)
2. I get the error message (as can be seen 7s after the start of the video) with other entities as well. Cannot find a reason for this error message and what device could be next ... no idea so far.
3. Switching the lamp to "off" results in bouncing back into the "on" state even if the lamp is "off" in reality. 
Maybe this helps in further investigation.

Please disregard the sound of the video :-)

Thanks F

https://github.com/Mat931/digitalstrom-homeassistant/assets/161063669/db589b03-d57f-4483-be0f-ad0b8b6a02be

**Comment by Mat931** [comment]
Hi @fraankkr 

The second time you turn on the switch you get a "response timeout" error message from the dSS. This could mean that it's still busy processing the first request. Sending commands to the lamp over the power wires is really slow and there's not much we can do about it.
Maybe I can fix the issue with the switch bouncing back, I have to take a closer look why that happens.

Thanks,
Mat

**Comment by fraankkr** [comment]
Hi Mat, thanks for your answer and the promise to look further into the new development. I compared different integrations with digitalstrom over the last few months and years (openhab, iobroker and homeassistant) and found that the integration and responsivness with iobroker works best till now. I would actually like to stick with your solution, cause I find that the integration and ease of use is best with homeassistant. I would be perfect if you would work further on this topic. I send you some screenshots from the latest error messages and the log file for your bug fixing. In case you want to test somethinge with our installation we can try this. Thanks for your help, Frank


![image](https://github.com/Mat931/digitalstrom-homeassistant/assets/161063669/939e6c81-f3eb-454d-9bbb-1b0e7bb41a1a)
![image](https://github.com/Mat931/digitalstrom-homeassistant/assets/161063669/7241e90e-b6ea-4b6f-9ad4-3299cc7664ea)
![image](https://github.com/Mat931/digitalstrom-homeassistant/assets/161063669/7ebaf9ec-cac8-4c58-9b1e-ccbb216d966d)
[home-assistant_2024-02-29T06-03-46.755Z.log](https://github.com/Mat931/digitalstrom-homeassistant/files/14443583/home-assistant_2024-02-29T06-03-46.755Z.log)

**Comment by Mat931** [comment]
Thank you for the log! It looks like your dSS is overloaded and takes a lot of time to respond to Home Assistant. I'm not familiar with the difference between different dSS hardware but maybe replacing it with a new one would give you better performance.

**Comment by fraankkr** [comment]
Hi, wäre ne Möglichkeit, allerdings gab es bei Openhab oder Iobroker mit der gleichen Installation weniger Probleme. Ist die Technik hier bzgl. der Anbindung komplett anders gelöst ?

**Comment by Mat931** [comment]
Bei Iobroker gibt es eine Queue (Warteschlange) für Anfragen an den Server, dass dieser weniger belastet wird. So etwas kann ich ja testweise in meine Integration einbauen, vielleicht ist das Problem damit schon gelöst.

**Comment by Mat931** [comment]
Ja, die Last kann dadurch zunehmen. Momentan lasse ich Home Assistant das Aktualisieren der Sensoren übernehmen. Da kann es sein dass viele Sensoren gleichzeitig abgefragt werden.

**Comment by fraankkr** [comment]
ok, ich schau mal, ob ich ein Upgrade des DSS Servers vornehmen kann. Mal sehen, was dann passiert. Gruss und Danke für die Arbeit. Melde mich wieder ...

**Comment by fraankkr** [comment]
Hat es eigentlich einen Grund warum Szenen aus Digitalstrom Umgebungen nicht direkt aufgerufen werden können ? Kommt das noch vielleicht ?

Grüße

**Comment by Mat931** [comment]
Das kommt noch :)

**Comment by fraankkr** [comment]
alles klar :-), so ist´s wenn man Vorfreude hat ....

**Comment by fraankkr** [comment]
Wäre es eine Möglichkeit das Intervall der Abfrage für den user als Variable zu hinterlegen ? Wie dies z.B: in IOBroker der Fall ist ? 

![image](https://github.com/Mat931/digitalstrom-homeassistant/assets/161063669/296ee286-8023-4b91-a4b5-06ad992297df)

**Comment by Mat931** [comment]
@fraankkr Das wäre zwar möglich, würde aber das Problem dass alle Geräte manchmal gleichzeitig abgefragt werden nicht lösen.
Ich denke die Warteschlange ist hier die vielversprechendste Lösung. Dafür muss aber leider ein großer Teil der Integration neu geschrieben werden.

**Comment by Mat931** [comment]
Hi @fraankkr ,
ich habe in Version 0.0.9 die Anzahl der gleichzeitigen Anfragen auf 1 pro Plattform begrenzt. Damit sollte es jetzt hoffentlich weniger Timeout-Fehler geben. Übrigens können jetzt auch Bereichs-Szenen aufgerufen werden.
Grüße

**Comment by rvdgaag** [comment]
@Mat931 did you use the cached values of the dSS with the latest API ? This prevents the dSS from crashing caused by overloading the dSS-bus. I can share the API if you need it !
